### PR TITLE
qtkeychain: 0.9.1 -> 0.12.0

### PR DESCRIPTION
**Pull request by symphorien**
this is necessary for nheko to work correctly

qt4 support is removed as it does not compile anymore, and there are no
reverse dependencies in nixpkgs according to nix-review.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132457` at 2f890d14 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1ed405a8c81043645021eb88729465ac">libsForQt514.akonadi-calendar</a></li>
    <li><a href="https://gist.github.com/527571c82002581839119230ebac671c">libsForQt514.calendarsupport</a></li>
    <li><a href="https://gist.github.com/432ae270f6554cd0eda4d4238f78a58a">libsForQt514.eventviews</a></li>
    <li><a href="https://gist.github.com/6288514466be2de526a448d192f9f200">libsForQt514.incidenceeditor</a></li>
    <li><a href="https://gist.github.com/405552bcc928457aacdf19bbee3b0efb">libsForQt514.kmailtransport</a></li>
    <li><a href="https://gist.github.com/3a4b38a4143a5e7fb766caab580dafa0">libsForQt514.libgravatar</a></li>
    <li><a href="https://gist.github.com/10bf38218a44c5b12fdc0a475704604a">libsForQt514.libkdepim</a></li>
    <li><a href="https://gist.github.com/702342e728c838eb4fa42248a0dd58da">libsForQt514.mailimporter</a></li>
    <li><a href="https://gist.github.com/b06ccd3693442b64095f4b1f1ff3ebae">libsForQt514.pimcommon</a></li>
  </ul>
</details>
<details>
  <summary>85 packages skipped due to time constraints:</summary>
  <ul>
    <li>calligra</li>
    <li>drawpile</li>
    <li>kstars</li>
    <li>libsForQt5.akonadi-calendar-tools (libsForQt515.akonadi-calendar-tools ,plasma5Packages.akonadi-calendar-tools)</li>
    <li>libsForQt5.akonadi-import-wizard (libsForQt515.akonadi-import-wizard ,plasma5Packages.akonadi-import-wizard)</li>
    <li>libsForQt5.akonadiconsole (libsForQt515.akonadiconsole ,plasma5Packages.akonadiconsole)</li>
    <li>libsForQt5.akregator (libsForQt515.akregator ,plasma5Packages.akregator)</li>
    <li>libsForQt5.kaddressbook (libsForQt515.kaddressbook ,plasma5Packages.kaddressbook)</li>
    <li>libsForQt5.kalarm (libsForQt515.kalarm ,plasma5Packages.kalarm)</li>
    <li>libsForQt5.kdepim-addons (libsForQt515.kdepim-addons ,plasma5Packages.kdepim-addons)</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>15 packages built successfully:</summary>
  <ul>
    <li>libsForQt5.akonadi-calendar (libsForQt515.akonadi-calendar ,plasma5Packages.akonadi-calendar)</li>
    <li>libsForQt5.calendarsupport (libsForQt515.calendarsupport ,plasma5Packages.calendarsupport)</li>
    <li>libsForQt5.eventviews (libsForQt515.eventviews ,plasma5Packages.eventviews)</li>
    <li>libsForQt5.incidenceeditor (libsForQt515.incidenceeditor ,plasma5Packages.incidenceeditor)</li>
    <li>libsForQt5.kldap (libsForQt515.kldap ,plasma5Packages.kldap)</li>
    <li>libsForQt5.kmailtransport (libsForQt515.kmailtransport ,plasma5Packages.kmailtransport)</li>
    <li>libsForQt5.libgravatar (libsForQt515.libgravatar ,plasma5Packages.libgravatar)</li>
    <li>libsForQt5.libkdepim (libsForQt515.libkdepim ,plasma5Packages.libkdepim)</li>
    <li>libsForQt5.mailimporter (libsForQt515.mailimporter ,plasma5Packages.mailimporter)</li>
    <li>libsForQt5.pimcommon (libsForQt515.pimcommon ,plasma5Packages.pimcommon)</li>
    <li>libsForQt5.qtkeychain (libsForQt515.qtkeychain ,plasma5Packages.qtkeychain)</li>
    <li>libsForQt512.qtkeychain</li>
    <li>libsForQt514.kldap</li>
    <li>libsForQt514.qtkeychain</li>
    <li>restique</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/qtkeychain/default.nix:36:3:
```
   |
36 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/libraries/qtkeychain/default.nix:9:3:
```
  |
9 |   name = "qtkeychain-${version}";
  |   ^
```

Near pkgs/development/libraries/qtkeychain/default.nix:10:3:
```
   |
10 |   version = "0.12.0";            # verify after nix-build with `grep -R "set(PACKAGE_VERSION " result/`
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by symphorien** [comment]
It looks like the 9 packages that failed to build are marked as broken.

File: pkgs/development/libraries/qtkeychain/default.nix [hunk 1]
```diff
@@ -1,27 +1,24 @@
-{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, qt4 ? null
-, withQt5 ? false, qtbase ? null, qttools ? null
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, qtbase, qttools
 , darwin ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qtkeychain/default.nix:2]:
```suggestion
, CoreFoundation, Security
```

File: pkgs/development/libraries/qtkeychain/default.nix [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
 , libsecret
 }:
 
-assert withQt5 -> qtbase != null;
-assert withQt5 -> qttools != null;
 assert stdenv.isDarwin -> darwin != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qtkeychain/default.nix:7]:
```suggestion
```

File: pkgs/development/libraries/qtkeychain/default.nix [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
     repo = "qtkeychain";
     rev = "v${version}";
-    sha256 = "0h4wgngn2yl35hapbjs24amkjfbzsvnna4ixfhn87snjnq5lmjbc"; # v0.9.1
+    sha256 = "0gi1nx4bcc1vwfw41cif3xi2i59229vy0kc2r5959d8n6yv31kfr"; # v0.9.1
   };
 
   dontWrapQtApps = true;
 
-  patches = (if withQt5 then [] else [ ./0001-Fixes-build-with-Qt4.patch ]) ++ (if stdenv.isDarwin then [ ./0002-Fix-install-name-Darwin.patch ] else []);
+  patches = lib.optional stdenv.isDarwin ./0002-Fix-install-name-Darwin.patch;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qtkeychain/default.nix:21]: Does it break if we apply it all the time?

File: pkgs/development/libraries/qtkeychain/default.nix [hunk 4]
```diff
@@ -30,7 +27,7 @@ stdenv.mkDerivation rec {
   ;
 
   buildInputs = lib.optionals (!stdenv.isDarwin) [ libsecret ]
-    ++ (if withQt5 then [ qtbase qttools ] else [ qt4 ])
+    ++ [ qtbase qttools ]
     ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qtkeychain/default.nix:31]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -18140,8 +18140,6 @@ in
   qtEnv = qt5.env;
   qt5Full = qt5.full;
 
-  qtkeychain = callPackage ../development/libraries/qtkeychain { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:18143]: Please add an alias.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/qtkeychain/default.nix (hunk 3 above)
Comment (symphorien) [reply pkgs/development/libraries/qtkeychain/default.nix]: yes, good idea, thanks

File: pkgs/development/libraries/qtkeychain/default.nix (hunk 4 above)
Comment (symphorien) [reply pkgs/development/libraries/qtkeychain/default.nix]: done

File: pkgs/development/libraries/qtkeychain/default.nix (hunk 2 above)
Comment (symphorien) [reply pkgs/development/libraries/qtkeychain/default.nix]: done

File: pkgs/development/libraries/qtkeychain/default.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/development/libraries/qtkeychain/default.nix]: done

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (symphorien) [reply pkgs/top-level/all-packages.nix:18143]: done

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 132457` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
60 packages marked as broken and skipped:
</summary>
<ul>
<li>libsForQt512.akonadi-calendar</li>
<li>libsForQt512.akonadi-calendar-tools</li>
<li>libsForQt512.akonadi-import-wizard</li>
<li>libsForQt512.akonadiconsole</li>
<li>libsForQt512.akregator</li>
<li>libsForQt512.calendarsupport</li>
<li>libsForQt512.eventviews</li>
<li>libsForQt512.incidenceeditor</li>
<li>libsForQt512.kaddressbook</li>
<li>libsForQt512.kalarm</li>
<li>libsForQt512.kdepim-addons</li>
<li>libsForQt512.kdepim-runtime</li>
<li>libsForQt512.kio-gdrive</li>
<li>libsForQt512.kldap</li>
<li>libsForQt512.kmail</li>
<li>libsForQt512.kmail-account-wizard</li>
<li>libsForQt512.kmailtransport</li>
<li>libsForQt512.knotes</li>
<li>libsForQt512.kontact</li>
<li>libsForQt512.korganizer</li>
<li>libsForQt512.libgravatar</li>
<li>libsForQt512.libkdepim</li>
<li>libsForQt512.libksieve</li>
<li>libsForQt512.mailcommon</li>
<li>libsForQt512.mailimporter</li>
<li>libsForQt512.mbox-importer</li>
<li>libsForQt512.messagelib</li>
<li>libsForQt512.pim-data-exporter</li>
<li>libsForQt512.pim-sieve-editor</li>
<li>libsForQt512.pimcommon</li>
<li>libsForQt514.akonadi-calendar</li>
<li>libsForQt514.akonadi-calendar-tools</li>
<li>libsForQt514.akonadi-import-wizard</li>
<li>libsForQt514.akonadiconsole</li>
<li>libsForQt514.akregator</li>
<li>libsForQt514.calendarsupport</li>
<li>libsForQt514.eventviews</li>
<li>libsForQt514.incidenceeditor</li>
<li>libsForQt514.kaddressbook</li>
<li>libsForQt514.kalarm</li>
<li>libsForQt514.kdepim-addons</li>
<li>libsForQt514.kdepim-runtime</li>
<li>libsForQt514.kio-gdrive</li>
<li>libsForQt514.kldap</li>
<li>libsForQt514.kmail</li>
<li>libsForQt514.kmail-account-wizard</li>
<li>libsForQt514.kmailtransport</li>
<li>libsForQt514.knotes</li>
<li>libsForQt514.kontact</li>
<li>libsForQt514.korganizer</li>
<li>libsForQt514.libgravatar</li>
<li>libsForQt514.libkdepim</li>
<li>libsForQt514.libksieve</li>
<li>libsForQt514.mailcommon</li>
<li>libsForQt514.mailimporter</li>
<li>libsForQt514.mbox-importer</li>
<li>libsForQt514.messagelib</li>
<li>libsForQt514.pim-data-exporter</li>
<li>libsForQt514.pim-sieve-editor</li>
<li>libsForQt514.pimcommon</li>
</ul>
</details>
<details>
<summary>
49 packages built:
</summary>
<ul>
<li>calligra</li>
<li>drawpile</li>
<li>kstars</li>
<li>libsForQt5.akonadi-calendar (libsForQt515.akonadi-calendar ,plasma5Packages.akonadi-calendar)</li>
<li>libsForQt5.akonadi-calendar-tools (libsForQt515.akonadi-calendar-tools ,plasma5Packages.akonadi-calendar-tools)</li>
<li>libsForQt5.akonadi-import-wizard (libsForQt515.akonadi-import-wizard ,plasma5Packages.akonadi-import-wizard)</li>
<li>libsForQt5.akonadiconsole (libsForQt515.akonadiconsole ,plasma5Packages.akonadiconsole)</li>
<li>libsForQt5.akregator (libsForQt515.akregator ,plasma5Packages.akregator)</li>
<li>libsForQt5.calendarsupport (libsForQt515.calendarsupport ,plasma5Packages.calendarsupport)</li>
<li>libsForQt5.eventviews (libsForQt515.eventviews ,plasma5Packages.eventviews)</li>
<li>libsForQt5.incidenceeditor (libsForQt515.incidenceeditor ,plasma5Packages.incidenceeditor)</li>
<li>libsForQt5.kaddressbook (libsForQt515.kaddressbook ,plasma5Packages.kaddressbook)</li>
<li>libsForQt5.kalarm (libsForQt515.kalarm ,plasma5Packages.kalarm)</li>
<li>libsForQt5.kdepim-addons (libsForQt515.kdepim-addons ,plasma5Packages.kdepim-addons)</li>
<li>libsForQt5.kdepim-runtime (libsForQt515.kdepim-runtime ,plasma5Packages.kdepim-runtime)</li>
<li>libsForQt5.kio-gdrive (libsForQt515.kio-gdrive ,plasma5Packages.kio-gdrive)</li>
<li>libsForQt5.kldap (libsForQt515.kldap ,plasma5Packages.kldap)</li>
<li>libsForQt5.kmail (libsForQt515.kmail ,plasma5Packages.kmail)</li>
<li>libsForQt5.kmail-account-wizard (libsForQt515.kmail-account-wizard ,plasma5Packages.kmail-account-wizard)</li>
<li>libsForQt5.kmailtransport (libsForQt515.kmailtransport ,plasma5Packages.kmailtransport)</li>
<li>libsForQt5.knotes (libsForQt515.knotes ,plasma5Packages.knotes)</li>
<li>libsForQt5.kontact (libsForQt515.kontact ,plasma5Packages.kontact)</li>
<li>libsForQt5.korganizer (libsForQt515.korganizer ,plasma5Packages.korganizer)</li>
<li>libsForQt5.libgravatar (libsForQt515.libgravatar ,plasma5Packages.libgravatar)</li>
<li>libsForQt5.libkdepim (libsForQt515.libkdepim ,plasma5Packages.libkdepim)</li>
<li>libsForQt5.libksieve (libsForQt515.libksieve ,plasma5Packages.libksieve)</li>
<li>libsForQt5.mailcommon (libsForQt515.mailcommon ,plasma5Packages.mailcommon)</li>
<li>libsForQt5.mailimporter (libsForQt515.mailimporter ,plasma5Packages.mailimporter)</li>
<li>libsForQt5.mbox-importer (libsForQt515.mbox-importer ,plasma5Packages.mbox-importer)</li>
<li>libsForQt5.messagelib (libsForQt515.messagelib ,plasma5Packages.messagelib)</li>
<li>libsForQt5.pim-data-exporter (libsForQt515.pim-data-exporter ,plasma5Packages.pim-data-exporter)</li>
<li>libsForQt5.pim-sieve-editor (libsForQt515.pim-sieve-editor ,plasma5Packages.pim-sieve-editor)</li>
<li>libsForQt5.pimcommon (libsForQt515.pimcommon ,plasma5Packages.pimcommon)</li>
<li>libsForQt5.qtkeychain (libsForQt515.qtkeychain ,plasma5Packages.qtkeychain)</li>
<li>libsForQt512.qtkeychain</li>
<li>libsForQt514.qtkeychain</li>
<li>mirage-im</li>
<li>mixxx</li>
<li>mudlet</li>
<li>neochat</li>
<li>nextcloud-client</li>
<li>nheko</li>
<li>owncloud-client</li>
<li>qgis</li>
<li>qgis-unwrapped</li>
<li>quaternion</li>
<li>restique</li>
<li>trojita</li>
<li>zanshin</li>
</ul>
</details>

**Pull request merged**